cmd/kube-proxy/app/server_others.go

```diff
@@ -157,22 +155,20 @@ func newProxyServer(
 	var ipt [2]utiliptables.Interface
 	dualStack := true // While we assume that node supports, we do further checks below
 
-	if proxyMode != proxyconfigapi.ProxyModeUserspace {
-		// Create iptables handlers for both families, one is already created
-		// Always ordered as IPv4, IPv6
-		if primaryProtocol == utiliptables.ProtocolIPv4 {
-			ipt[0] = iptInterface
-			ipt[1] = utiliptables.New(execer, utiliptables.ProtocolIPv6)
-		} else {
-			ipt[0] = utiliptables.New(execer, utiliptables.ProtocolIPv4)
-			ipt[1] = iptInterface
-		}
+	// Create iptables handlers for both families, one is already created
+	// Always ordered as IPv4, IPv6
+	if primaryProtocol == utiliptables.ProtocolIPv4 {
+		ipt[0] = iptInterface
+		ipt[1] = utiliptables.New(execer, utiliptables.ProtocolIPv6)
```

Comment:
The iptinterface initialization stuff could be rewritten to make more sense now. eg, you can remove `iptInterface` (and `ProxyServer.IptInterface`) and only have `ipt`, and you don't even need to figure out `primaryProtocol` unless `dualStack` gets set `false`

(Maybe do this as an additional refactoring commit to make it more obvious that this commit is just removing userspace and nothign else.)

(Oh, I see you left a "todo" about EndpointSlices so maybe you're planning on doing another PR later?)

Reply:
@danwinship I would prefer to refactor this part on a separated commit and let this interface as is for now, wdyt?

On this PR already cleaned the `UseEndpointSlices` and forced the modes to use it by default.